### PR TITLE
Task #58542 [LiconKA] Function-2 : Play Recorded File

### PR DESCRIPTION
**Pull request by liconrepo**

### CODE REVIEW

File: app/src/main/java/com/licon/soundstudio/ui/fragment/RecordFragment.java [hunk 1]
```diff
@@ -18,28 +21,34 @@
...
+import com.licon.soundstudio.util.T;
 import com.licon.soundstudio.util.UIUtil;
 
 import java.io.IOException;
 
 public class RecordFragment extends Fragment implements View.OnClickListener {
 
     private static final int REPEAT_INTERVAL = 40;
+    private static final float ALPHA_MIN = (float) 0.4;
+    private static final float ALPHA_MAX = (float) 1.0;
```
Comment (b1acKr0se) [inline app/src/main/java/com/licon/soundstudio/ui/fragment/RecordFragment.java]: add f at the end of the number so you don't need to cast, e.g 1.0f

File: app/src/main/java/com/licon/soundstudio/util/UIUtil.java [hunk 2]
```diff
@@ -24,4 +28,13 @@ public static Animation getAnimationRotateReverse(Context context) {
         Animation animRotateRev = AnimationUtils.loadAnimation(context, R.anim.anim_rotate_reverse);
         return animRotateRev;
     }
+
+    @TargetApi(Build.VERSION_CODES.JELLY_BEAN)
+    public static void addResToImageButton(ImageButton imageButton, int resId) {
+        if(AppData.CURRENT_SDK_VERSION < AppData.getJellyBean()) {
+            imageButton.setImageResource(resId);
+        } else {
+            imageButton.setImageResource(resId);
```
Comment (b1acKr0se) [inline app/src/main/java/com/licon/soundstudio/util/UIUtil.java]: why do if and else statement have the same content?

File: app/src/main/java/com/licon/soundstudio/util/T.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package com.licon.soundstudio.util;
+
+/**
+ * Created by FRAMGIA\khairul.alam.licon on 1/4/16.
+ */
+import android.content.Context;
+import android.widget.Toast;
+
+public class T {
```
Comment (b1acKr0se) [inline app/src/main/java/com/licon/soundstudio/util/T.java]: Class name does not make sense

File: app/src/main/java/com/licon/soundstudio/ui/widget/RecordPlayer.java [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+
+    public DialogInterface.OnClickListener mOnClickListener;
+
+    private MediaPlayer mMediaPlayer;
+    private double mInitTime = 0;
+    private double mEndTime = 0;
+    private int mForwardTime = 5000;
+    private int mRewindTime = 5000;
+    private int mOneTimeOnly = 0;
+    private String mFileUrl = null;
```
Comment (b1acKr0se) [inline app/src/main/java/com/licon/soundstudio/ui/widget/RecordPlayer.java]: again, dont have to initialise

File: app/src/main/java/com/licon/soundstudio/ui/widget/RecordPlayer.java [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+    private Context mContext;
+    private Activity mActivity;
+    private boolean isPlaying = false;
```
Comment (b1acKr0se) [inline app/src/main/java/com/licon/soundstudio/ui/widget/RecordPlayer.java]: this as well

File: app/src/main/java/com/licon/soundstudio/ui/fragment/RecordFragment.java [hunk 6]
```diff
@@ -18,28 +21,34 @@
...
     private FrequencyView mFrequencyView;
     private CassetteView mCassetteView;
+    private Context mContext;
 
     private MediaRecorder mMediaRecorder;
     private String mOutFile;
     private boolean isRecording;
+    private long mTimeWhenStopped = 0;
 
     private Handler handler;
```
Comment (b1acKr0se) [inline app/src/main/java/com/licon/soundstudio/ui/fragment/RecordFragment.java]: rename

File: app/src/main/java/com/licon/soundstudio/util/UIUtil.java [hunk 7]
```diff
@@ -24,4 +28,12 @@ public static Animation getAnimationRotateReverse(Context context) {
         Animation animRotateRev = AnimationUtils.loadAnimation(context, R.anim.anim_rotate_reverse);
         return animRotateRev;
     }
+
+    @TargetApi(Build.VERSION_CODES.JELLY_BEAN)
```
Comment (b1acKr0se) [inline app/src/main/java/com/licon/soundstudio/util/UIUtil.java]: what is this for then?

File: app/src/main/java/com/licon/soundstudio/constants/AppData.java [hunk 8]
```diff
@@ -20,5 +21,11 @@
     public static final int RECORDER_AUDIO_ENCODER = MediaRecorder.OutputFormat.AMR_NB;
     // file & folder to save recorded audio
     public static String APP_FOLDER_NAME = "Sound Studio";
+    public static String FILE_PREFIX = "SS_";
     public static String FILE_FORMAT_3GP = ".3gp";
+    // sdk data
+    public static final int CURRENT_SDK_VERSION = android.os.Build.VERSION.SDK_INT;
+    public static int getJellyBean() {
+        return Build.VERSION_CODES.JELLY_BEAN;
```
Comment (b1acKr0se) [inline app/src/main/java/com/licon/soundstudio/constants/AppData.java]: Have you used these variables in your app? Or you intent to use it in the future?

File: app/src/main/java/com/licon/soundstudio/ui/fragment/RecordFragment.java [hunk 9]
```diff
@@ -101,42 +125,77 @@ public void onClick(View view) {
...
             } catch (IOException e) {
                 e.printStackTrace();
             }
             mButtonRecord.setEnabled(false);
             mButtonDone.setEnabled(true);
+            mButtonRecord.setAlpha(ALPHA_MIN);
+            mButtonDone.setAlpha(ALPHA_MAX);
             mViewRedDotLight.startAnimation(UIUtil.getAnimationBlink(getActivity().getApplicationContext()));
-            Snack.show(getResources().getString(R.string.msg_record_on), mRelativeLayoutRecord);
+            AppToast.show(mContext, mContext.getResources().getString(R.string.msg_record_on));
```
Comment (b1acKr0se) [inline app/src/main/java/com/licon/soundstudio/ui/fragment/RecordFragment.java]: `Context` has a `getString` method so the `getResource()` is unnecessary. I see you did this in a lot of occasions, please fix them.

File: app/src/main/java/com/licon/soundstudio/ui/fragment/RecordFragment.java [hunk 10]
```diff
@@ -101,42 +125,77 @@ public void onClick(View view) {
...
+                mChronometer.setVisibility(View.VISIBLE);
+                mChronometer.setBase(SystemClock.elapsedRealtime() + mTimeWhenStopped);
                 mCassetteView.startCassetteAnimation(CassetteView.ROTATE_FRONT);
                 mViewRedDotLight.setVisibility(View.VISIBLE);
                 isRecording = false;
-                handler.post(updateVisualizer);
+                mHandler.post(updateVisualizer);
             } catch (IllegalStateException e) {
                 e.printStackTrace();
             } catch (IOException e) {
```
Comment (b1acKr0se) [inline app/src/main/java/com/licon/soundstudio/ui/fragment/RecordFragment.java]: can just catch multiple exceptions in one catch block

**Comment by b1acKr0se** [comment]
OK

**Comment by nvmanh** [comment]
OK!

**Pull request merged**